pandas/_libs/tslibs/fields.pyx

```diff
@@ -253,8 +254,7 @@ def get_start_end_field(
         # month of year. Other offsets use month, startingMonth as ending
         # month of year.
 
-        if (freqstr[0:2] in ["MS", "QS", "YS"]) or (
-                freqstr[1:3] in ["MS", "QS", "YS"]):
+        if re.split("[0-9]*", freqstr, maxsplit=1)[1][0:2] in ["MS", "QS", "YS"]:
```

Comment:
i know this is still in draft, but is there any existing function that can be re-used here? it seems like a fairly common thing to get the frequency out of a string, I'm sure there's other places that do it - can something be reused? I don't think it's really feasible to have a regex each time this needs doing